ovos_backend_client/backends/base.py

```diff
@@ -423,16 +359,48 @@ def oauth_get_token(self, dev_cred):
         raise NotImplementedError()
 
     # Dataset API
-    @abc.abstractmethod
-    def dataset_upload_wake_word(self, audio, params):
+    def dataset_upload_wake_word(self, audio, params, upload_url=None):
         """ upload wake word sample - url can be external to backend"""
-        raise NotImplementedError()
+        byte_data = audio.get_wav_data()
+        upload_url = upload_url or Configuration().get("listener", {}).get("wake_word_upload", {}).get("url")
+        if upload_url:
+            # upload to arbitrary server
+            ww_files = {
+                'audio': BytesIO(byte_data),
+                'metadata': StringIO(json.dumps(params))
+            }
+            return self.post(upload_url, files=ww_files)
```

Comment:
mentioned the header problem earlier. we _can not_ send _any_ `content_type` . The correct type will be chosen.

The files have to be sent as `multipart/form-data`. The problem with that is that this type has a boundary param associated, which is generated automatically by `requests`, but not if we predefine it as `multipart/form-data`

[This issue](https://github.com/postmanlabs/postman-app-support/issues/191) is about postman, but the same applies here

Reply:
this should be a separate PR coordinated with a companion PR in personal backend